File: 🟢 Easy/1572 - Matrix Diagonal Sum/_1572MatrixDiagonalSum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        ans = 0

        for i in range(0, n):
            ans += mat[i][i]

            if i != n - 1 - i:
                ans += mat[i][n - i - 1]

        return ans

File: 🟢 Easy/1572 - Matrix Diagonal Sum/_1572MatrixDiagonalSum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n = mat.length;
        int sum = 0;

        for (int i = 0; i < n; i++) {
            // pd
            sum += mat[i][i];

            // sd + overlapping condition
            if (i != n - 1 - i) {
                sum += mat[i][n - i - 1];
            }
        }

        return sum;
    }
}